editor/plugins/node_3d_editor_plugin.cpp

```diff
@@ -5179,6 +5179,36 @@ void Node3DEditorViewport::_set_lock_view_rotation(bool p_lock_rotation) {
 	}
 }
 
+void Node3DEditorViewport::_add_advanced_debug_draw_mode_item(PopupMenu *p_popup, const String &p_name, int p_value, SupportedRenderingMethods p_rendering_methods, const String &p_tooltip) {
+	display_submenu->add_radio_check_item(p_name, p_value);
+
+	bool disabled = false;
+	String disabled_tooltip;
+	switch (p_rendering_methods) {
+		case SupportedRenderingMethods::ALL:
+			break;
+		case SupportedRenderingMethods::FORWARD_PLUS_MOBILE:
+			disabled = OS::get_singleton()->get_current_rendering_method() == "gl_compatibility";
+			disabled_tooltip = TTR("This debug draw mode is not supported when using the Compatibility rendering method.");
+			break;
+		case SupportedRenderingMethods::FORWARD_PLUS:
+			disabled = OS::get_singleton()->get_current_rendering_method() == "gl_compatibility" || OS::get_singleton()->get_current_rendering_method() == "mobile";
```

Comment:
This does the same thing in practice, but it's less semantically correct. It may matter in the future if we have a Cinematic rendering method (as mentioned in [this gist](https://gist.github.com/reduz/c5769d0e705d8ab7ac187d63be0099b5)), as this rendering method would likely support all debug draw modes.